src/util/sha256_openssl.c

```diff
@@ -11,38 +11,39 @@
 #include "sha256.h"
 #include "util.h"
 
-#include <openssl/sha.h>
+#include <openssl/evp.h>
 
 struct SHA256State {
-	SHA256_CTX context;
+	EVP_MD_CTX *context;
```

Comment:
>you can just embed it

Forgive my ignorance, but I'm not quite sure what that looks like in C.

Reply:
The same way it does in the original code, you just store a `EVP_MD_CTX context` instead of a `EVP_MD_CTX *context` in the struct, and use `&st->context` when you need a pointer to it.

Reply:
Ah, so that was the first thing I tried. It complains with `error: field has incomplete type 'EVP_MD_CTX' (aka 'struct evp_md_ctx_st')`, hence why I went for the simpler option of using a pointer.

Reply:
I can confirm it works on Debian 11/Ubuntu 20.04, which both ship with OpenSSL 1.1.1. I'll also update the docs to reflect this.